packages/inter-protocol/src/vaultFactory/vaultManager.js

```diff
@@ -708,12 +763,343 @@ export const prepareVaultManagerKit = (
         getPublicFacet() {
           return this.facets.collateral;
         },
+
+        lockOraclePrices() {
+          const { state, facets } = this;
+          const { self } = facets;
+
+          // XXX: 'twould be better if collateralQuote were updated lazily when it
+          // changes by more than X%, and we use a cached value.  see #6946
+          return E.when(self.getCollateralQuote(), quote => {
+            trace(`lockPrice`, getAmountIn(quote), getAmountOut(quote));
+            // @ts-expect-error declared PriceQuote | undefined. TS thinks undefined.
+            state.lockedQuote = quote;
+            return quote;
+          });
+        },
+        async liquidateVaults(auctionPF) {
+          const { state, facets } = this;
+          const { lockedQuote, compoundedInterest } = state;
+          const oraclePriceAtStartP = facets.self.getCollateralQuote();
+
+          assert(factoryPowers && prioritizedVaults && zcf);
+          lockedQuote ||
+            Fail`Must have locked a quote before liquidating vaults.`;
+          assert(lockedQuote); // redundant with previous line
+
+          const crKey = normalizedCrKey(lockedQuote, compoundedInterest);
+
+          trace(
+            `Liquidating vaults worse than`,
+            normalizedCr(lockedQuote, compoundedInterest),
+          );
+          const {
+            totalDebt = zeroDebt,
+            totalCollateral = zeroCollateral,
```

Comment:
these default vals are a little surprising. consider passing `zeroDebt` and `zeroCollateral` to `getLiquidatableVaults`. That would also be an earlier check on the brands matching. Right now if the prioritizedVaults is for a different pair of brands but it's empty, then this function will return and the totalDebt could be a different brand.

Maybe `makePrioritizedVaults` should take `debtBrand` and `collateralBrand` as params and check during `addVault` that they match. Then it could have getters for those brands and `getLiquidatableVaults` could know the zero amounts without any new params.

Reply:
I kinda like the second choice better, but vaults don't reveal their brands easily (at a quick glance), so I went with the first choice instead.